### ISSUE TITLE
LLVM 3.5 won't build with modern gcc

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#405]
LLVM 3.5 is a supported LLVM version in the `setup_env.py` script. However, LLVM 3.5 won't build with modern versions of gcc, unless this patch is applied to `include/llvm/ADT/IntrusiveRefCntPtr.h`:

```
--- IntrusiveRefCntPtr.h.old    2018-06-26 18:04:09.036217914 -0700
+++ IntrusiveRefCntPtr.h    2018-06-25 15:36:38.688578563 -0700
@@ -135,6 +135,7 @@
   template <typename T>
   class IntrusiveRefCntPtr {
     T* Obj;
+    template<typename U> friend class IntrusiveRefCntPtr;
 
   public:
     typedef T element_type;
```

`setup_env.py` could apply this patch automatically before building.

**Comment by elliottslaughter** [comment]
Should be fixed as of bac73690a046f50099be0d39b1cd2bf59d186c6d. Thanks!

**Comment by manopapad** [comment]
Fixed!
